### DISCUSSION TITLE
Gopro Labs Construction Time Lapse - powers on/off between shot intervals

### DISCUSSION THREAD

**Discussion by Pallenwht** [gopro/labs#183] [General]
I am about 2 months into an 8 month construction time lapse project. Using two Hero 8 cameras with the Gopro Labs “construction time lapse” QR programming. It works well, but the last time I swapped out the SD cards I found that one of the cameras didn’t record anything, two weeks worth of nothing, some problem with the memory card,  Since there was little activity at the site over the holiday season I brought both cameras home and ran the construction time lapse mode. The cameras start up fine when I activate with the QR printout card. The first time interval is fine — program powers the cameras off - off for 5 to 6 minutes, power on - capture the image - power off. BUT during the following intervals both cameras power back on and off every 10 seconds or so, but does not take a picture till the programmed time.  NOW! This is not a “project killer” because I am getting the shots I need for the project. MY CONCERN is battery power drain. Instead of the 3 to 4 days of battery usage, as indicated by the QR program, I am getting maybe 2 hours of operation. What am I missing here? The cameras don’t read the QR card correctly? The QR card to close or to far away from the camera when it activates? The QR card is upside down? The QR card is not lit properly? I get the little QR pop-up with the check mark image and the camera shuts off as expected. How can I fix this? Re-install the Gopro Labs firmware? Do a “factory reset” and start over?  Need some suggestions and advice here!!

**Comment by dnewman-gpsw** [comment]
Moved to discussions.

Please share the command you are running.

**Reply by Pallenwht** [reply]
Command?  I am using the Labs QR image for Construction Time Lapse That is the only command I know of.

**Reply by dnewman-gpsw** [reply]
The command is the text with the QR Code.  Please share.

**Reply by Pallenwht** [reply]
mPdP>07:59<16:30!332SQ-!08:00S!1R     Is this correct?  I'm kind of a novice here. ( "-" is supposed to be a tilde' character)

**Reply by dnewman-gpsw** [reply]
Yes that should work.  So it did work then stopped on both your HERO8 cameras?  What else changed?  

Start this QR command again, if it fails, stop, then grab the log file from the SD card from MISC/qrlog.txt and share that please.

**Reply by Pallenwht** [reply]
The program NEVER stops. It will capture the desired images in the required time frame, but it just keeps turning on and off between image captures which drains the batteries rapidly.  I will get that grlog.txt to you soon because I know that both cameras will react this way in just a few minutes.   I might be a little while getting back to you.  Thank You for your assistance here!!!!

**Reply by Pallenwht** [reply]
[qrlog.txt](https://github.com/gopro/labs/files/10377323/qrlog.txt)    I hope you can open this. Let me know if I need to cut and paste this.  I let the camera run from 1530 to 1600

**Reply by dnewman-gpsw** [reply]
Here is the issue in the Log file:
```
[2023-01-09 15:52:41] Shutdown and Sleep:138 seconds

11 seconds later

[2023-01-09 15:52:52] reset:rewake immediately

[2023-01-09 15:53:00] Labs:Main Start

[2023-01-09 15:53:00] Labs:Initialize

[2023-01-09 15:53:01] Woke with Command:[15:55m01]!SQ~!08:00S!1R

[2023-01-09 15:53:05] Shutdown and Sleep:114 seconds

11 seconds later

[2023-01-09 15:53:16] reset:rewake immediately
```

For some reason your camera is taking a long time to shutdown, and there is a timer that says if you have shutdown in 10 seconds, reboot.   

Format your media and factory reset your camera to see if it behaves better.

**Reply by Pallenwht** [reply]
Thank You!  I was thinking that the only recourse was to do a factory reset, but I hesitated because the reset would cancel my gopro registration and I don't know how much trouble that would cause. I had already reformatted the SD, but no resolve. I'll let you know how this turns out.  Anyway you have given me a procedure! Thanks!    PAUL

**Reply by Pallenwht** [reply]
I did the factory reset and media format but the problem is still present. Unusual thing about the factory reset - nothing changed. The camera went through the setup screens - select english as language etc. but the time and date was correct AND all my previous settings were still there. Is that the way the factory reset works?   Do you think that my QR code printed on glossy paper might be the problem? Possibly a light reflection causing some part of the QR printout to be missed or read incorrectly?

**Reply by Pallenwht** [reply]
I'll try printing the QR code on regular paper.

**Reply by dnewman-gpsw** [reply]
What previous settings?  Labs setting will remain, everything else should be reset with a factory reset.  To reset Labs setting (a good idea) use command !RESET.

**Reply by Pallenwht** [reply]
Yes! I was thinking that the factory reset would cancel the Gopro Labs firmware and I would just reload. But no. Your recommended solution of re-installing the Labs firmware is a good one, but I don't know what I am doing or how to do it. How do I generate a command !Reset?   I see that someone else is experiencing the identical problem that I am having with their time lapse..

**Reply by Pallenwht** [reply]
The discussion:   Timelapse issues #195 by logan12534

**Reply by dnewman-gpsw** [reply]
Never need to reinstall firmware. 
Here is the !RESET command (all commands are literal text, case sensitive.)  https://gopro.github.io/labs/control/set/?cmd=!RESET

**Reply by Pallenwht** [reply]
The reset QR you provided worked fine and I was able to fresh install the Labs firmware, but the problem still exists. Like I said in my initial post "this is not a project killer"! The program does what it is commanded to do and I am getting the needed images. My major concern is battery drainage. I equipped both Pelco enclosures with a 3.3AH battery and 12v to 5v DC to DC converter with the output connected to the camera. The battery is charged by a solar panel. My concern is that maybe after a few cloudy days and the camera on almost constantly the battery will drain and not provide power.     Is there somewhere that I can get a tutorial on programming with this type of code in order to better understand what I see in the log records? I used to program in Basic and assembly code, but I am not familiar with this type of code.   Thanks!

**Reply by dnewman-gpsw** [reply]
The commands are explained here https://gopro.github.io/labs/control/tech/  Then it is a lot of trail and error.

I suggest you follow this idea https://github.com/gopro/labs/discussions/188  more reliability and control of battery usage. https://github.com/gopro/labs/discussions/188

**Reply by Pallenwht** [reply]
OK!  Thanks for the advise!

**Reply by Pallenwht** [reply]
Thanks for steering me to the settings and action pages. Now I can understand what the line of the Construction TL code means.  Reading the previous logs it seems that the program gets behind for some reason and is trying to catch up by restarting often prior to the image capture. I have a couple of days before I MUST put the cameras back at the job site, So I am going to try create a code that works differently.  So now comes the "trial and error" process with the help of the !  I do have some syntax questions. What does the tilde symbol mean or do?  "R" means repeat the command, but what does the number (1) ahead or the R mean/do?  It's been a long time since I did any programming and this looks like it might be fun (or frustrating), but at least I'm learning something.  Thanks again!!!

**Reply by dnewman-gpsw** [reply]
The !1R the '1' is explained here https://gopro.github.io/labs/control/actions/
The ~ is documents but little harder to follow, it is the else in a conditional.

example in pseudo code:
```
if(time < 08:00) 
{
   set mode mV
   start capture
   end after 60s
}
else
{
   set mode photo
   start capture
}
repeat in an hour
```

In Labs:    `<08:00mV+!S+!60E~mP+!S!3600R`
